.changeset/sweet-shrimps-allow.md

```diff
@@ -0,0 +1,11 @@
+---
+"@salt-ds/core": minor
+---
+
+Add mergeProps to utils.
+
+This utility merges two prop objects according to the following rules:
```

Comment:
The detailed description feels more fit to either the doc site, or JsDoc comments.